### PR TITLE
add example demonstrating using a layout function to create a multi-

### PR DESCRIPTION
**Pull request by michaelbabyn**
page/tab app without suppress_callback_exceptions.

This PR address #172. I put the example under the other layout as a function example in live updates for now because I'm not sure where this would fit best.

@chriddyp, @cldougl maybe you could give me an idea of a better spot for this?

### CODE REVIEW

**Comment by T4rk1n** [comment]
@chriddyp  There's no deploy ?

File: tutorial/live_updates.py [hunk 1]
```diff
@@ -170,4 +170,110 @@ def serve_layout():
...
+import dash_html_components as html
+import dash_core_components as dcc
+from dash.dependencies import Input, Output, State
+
+import flask
+
+app = dash.Dash(__name__, external_stylesheets=['https://codepen.io/chriddyp/pen/bWLwgP.css'])
+
+layout_index = html.Div([
+    dcc.Tabs(id="tabs", value='tab-1', children=[
```
Comment (T4rk1n) [inline tutorial/live_updates.py]: I think an example with `dcc.Location` multi-page app would be better here, from what I've seen that pattern almost always suppress the exceptions. It is set in our multi-page examples https://dash.plot.ly/urls.

File: tutorial/live_updates.py [hunk 2]
```diff
@@ -170,4 +170,110 @@ def serve_layout():
...
+               State('input-2-state', 'value')])
+def update_output(n_clicks, input1, input2):
+    return ('The Button has been pressed {} times,'
+            'Input 1 is "{}",'
+            'and Input 2 is "{}"').format(n_clicks, input1, input2)
+
+
+# Tab 2 Callbacks
+@app.callback(dash.dependencies.Output('tab-2-display-value', 'children'),
+              [dash.dependencies.Input('tab-2-dropdown', 'value')])
```
Comment (T4rk1n) [inline tutorial/live_updates.py]: Replace the long form Output/Input with the imported classes.

File: tutorial/live_updates.py [hunk 3]
```diff
@@ -170,4 +170,110 @@ def serve_layout():
...
+For full validation, all components within your callback must therefore appear 
+in the initial layout of your app, and you will see an error if they do not.
+However, in the case of more complex Dash apps that involve dynamic modification 
+of the layout (such as multi-page apps), not every component appearing in your 
+callbacks will be included in the initial layout.
+
+Since this validation is done before flask has any request context, you can create 
+a layout function that checks `flask.has_request_context()` and returns a complete
+layout to the validator if there is no request context and returns the incomplete
+index layout otherwise:
```
Comment (T4rk1n) [inline tutorial/live_updates.py]: `:` typo ?

File: tutorial/live_updates.py [hunk 4]
```diff
@@ -170,4 +170,110 @@ def serve_layout():
...
 a unique `layout` every hour or every day and serve the computed `layout`
 from memory in between.
-''')]
+'''),
+
+          dcc.Markdown('''
+
+***
+
+## Dynamically Create Layout for Multi-page App Validation
```
Comment (T4rk1n) [inline tutorial/live_updates.py]: Dynamically create a layout for multi-page app validation.

**Comment by michaelbabyn** [comment]
@T4rk1n I updated this example to use dcc.Location instead of tabs to address your review

**Comment by michaelbabyn** [comment]
Yeah, you're right @T4rk1n and now that it uses urls it makes sense for it to be with the other multi-page app documentation.

**Pull request merged**